tests/tests_fabric/utilities/test_optimizer.py

```diff
@@ -1,36 +1,131 @@
+        fused_vals = [False] if device == "cpu" else [False, True]
+
+        for fused in fused_vals:
+            optimizer = torch.optim.Adam(model.parameters(), lr=0.1, fused=fused)
+            simple_training(optimizer, model, dataloader, loss_fn=nn.MSELoss())
+            if fused:
+                optimizer_on_device[device + "_fused_" + str(fused)] = optimizer
+            else:
+                optimizer_on_device[device] = optimizer
+
+    # Test _optimizer_to_device function
+    # Test cpu-->gpu, fused = False from CPU
+    opt_to_gpu = copy.deepcopy(optimizer_on_device["cpu"])
+    _optimizer_to_device(opt_to_gpu, gpu_device)
+    assert_opt_state_in_expected_location(opt_to_gpu, optimizer_on_device[gpu_device])
+
+    # Test gpu-->cpu, fused = False
+    opt_to_cpu = copy.deepcopy(optimizer_on_device[gpu_device])
+    _optimizer_to_device(opt_to_cpu, "cpu")
+    assert_opt_state_in_expected_location(opt_to_cpu, optimizer_on_device["cpu"])
+
+    # Test gpu-->cpu, fused = True
+    opt_to_cpu = copy.deepcopy(optimizer_on_device[gpu_device + "_fused_True"])
+    _optimizer_to_device(opt_to_cpu, "cpu")
+    assert_opt_state_in_expected_location(opt_to_cpu, optimizer_on_device["cpu"])
+
+    # Try from_dict
+    # These all pretend that we have an appropriate prototype, I don't think we can actually do this since
+    # all we may have is a CPU pickle
+    # For now, this is a future idea for _optimizer_to_device
+
+    if False:
+        # GPU prototypes
+        # Use from_dict with gpu prototype, fused = False
+        opt_cpu_dict = optimizer_on_device["cpu"].state_dict()
+        gpu_prototype = copy.deepcopy(optimizer_on_device[gpu_device])
+        gpu_prototype.load_state_dict(opt_cpu_dict)
+        print(opt_cpu_dict)
+        assert_opt_state_in_expected_location(gpu_prototype, optimizer_on_device[gpu_device])
+
+        # Use from_dict with gpu prototype, fused = True
+        opt_cpu_dict = optimizer_on_device["cpu"].state_dict()
+        gpu_prototype = copy.deepcopy(optimizer_on_device[gpu_device + "_fused_True"])
+        gpu_prototype.load_state_dict(opt_cpu_dict)
+        assert_opt_state_in_expected_location(
+            gpu_prototype, optimizer_on_device[gpu_device]
+        )  # fused=False from CPU, overrides prototype
+
+        # CPU prototypes
+        # Use from_dict with cpu prototype, fused = False
+        opt_gpu_dict = optimizer_on_device[gpu_device].state_dict()
+        cpu_prototype = copy.deepcopy(optimizer_on_device["cpu"])
+        cpu_prototype.load_state_dict(opt_gpu_dict)
+        assert_opt_state_in_expected_location(cpu_prototype, optimizer_on_device["cpu"])
+
+        # Use from_dict with cpu prototype, fused = True
+        opt_gpu_dict = optimizer_on_device[gpu_device + "_fused_True"].state_dict()
+        cpu_prototype = copy.deepcopy(optimizer_on_device["cpu"])
+        cpu_prototype.load_state_dict(opt_gpu_dict)  # This should give an error / refuse to allow fused = True
```

Comment:
FYI, for older versions of torch this should indeed be not allowed/would break. But since torch 2.4, there is a fused CPU Adam(W)/SGD/Adagrad, so fused=True on CPU for these optimizers would be valid.